tasks/static_quality_gates/experimental_gates.py

```diff
@@ -425,81 +525,634 @@ def _serialize_file_info(self, file_info: FileInfo) -> dict[str, Any]:
+
+    def _analyze_extracted_docker_layers(
+        self,
+        extract_dir: str,
+        debug: bool = False,
+    ) -> tuple[int, list[FileInfo]]:
+        """Analyze extracted crane pull tarball to get disk size and file inventory.
+
+        Handles hard links correctly by tracking inodes within each layer to avoid
+        counting the same file multiple times.
+        """
+        import json
+        import subprocess
+
+        total_disk_size = 0
+        all_files = {}  # Use dict to handle overwrites from different layers
+
+        try:
+            # Check if this is OCI format (has index.json and oci-layout)
+            index_path = os.path.join(extract_dir, "index.json")
+            oci_layout_path = os.path.join(extract_dir, "oci-layout")
+            manifest_path = os.path.join(extract_dir, "manifest.json")
+
+            if os.path.exists(index_path) and os.path.exists(oci_layout_path):
+                # OCI format
+                if debug:
+                    print("🔍 Detected OCI format")
+
+                with open(index_path) as f:
+                    index = json.load(f)
+
+                # Get the manifest digest from the index
+                manifest_digest = index["manifests"][0]["digest"]
+                manifest_blob_path = os.path.join(extract_dir, "blobs", manifest_digest.replace(":", "/"))
+
+                with open(manifest_blob_path) as f:
+                    manifest = json.load(f)
+
+                # Get layer digests from manifest
+                layer_digests = [layer["digest"] for layer in manifest.get("layers", [])]
+
+                # Convert to layer paths
+                layer_files = [os.path.join("blobs", digest.replace(":", "/")) for digest in layer_digests]
+
+            elif os.path.exists(manifest_path):
+                # Old tarball format
+                if debug:
+                    print("🔍 Detected Docker tarball format")
+
+                with open(manifest_path) as f:
+                    manifest = json.load(f)[0]  # Typically one image per tarball
+
+                layer_files = manifest.get("Layers", [])
+            else:
+                raise RuntimeError("Neither OCI format (index.json) nor tarball format (manifest.json) detected")
+
+            if debug:
+                print(f"🔍 Found {len(layer_files)} layers in manifest")
+
+            # Process each layer by extracting and walking files
+            for i, layer_file in enumerate(layer_files):
+                layer_path = os.path.join(extract_dir, layer_file)
+
+                if debug:
+                    print(f"📦 Processing layer {i + 1}/{len(layer_files)}: {layer_file}")
+
+                with tempfile.TemporaryDirectory() as layer_extract_dir:
+                    # Extract layer
+                    import shlex
+
+                    extract_command = (
+                        f"tar -xf {shlex.quote(layer_path)} -C {shlex.quote(layer_extract_dir)} 2>/dev/null"
```

Comment:
The `shlex` logic is incompatible with Windows and `tar` doesn't exist. Can we use the standard library's [tarfile](https://docs.python.org/3/library/tarfile.html) module?